### PR TITLE
chore: improve linting and update .gitignore paths

### PR DESCRIPTION
**Pull request by MaximDevoir**
This PR adds small changes that affect the testing of Create Probot App:

* Use ignore-paths from `.gitignore` for Prettier to avoid linting out-of-scope files (i.e. files generated from tests and package builds).
* Update the name of the testing folder from `tmp` to `test`.

### CODE REVIEW

**Comment by shaftoe** [comment]
It looks good to me, even though now I feel bad that I chose `test` as folder temporary name when the more appropriate `tmp` was already there.

**Comment by MaximDevoir** [comment]
If test is gitignored, it will probably ignore `test` from templates too, whoops!

File: .gitignore [hunk 1]
```diff
@@ -15,6 +12,7 @@ yarn.lock
 
 # Test-related generated folders
 coverage
+/test
```
Comment (shaftoe) [inline .gitignore:15]: How about this instead:

```diff
diff --git a/package.json b/package.json
index 090f7f9..d9911a3 100644
--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
   "scripts": {
     "test": "./script/test-all-templates.sh",
     "test:template": "./script/test-template.sh",
-    "test:clean": "rimraf test",
+    "test:clean": "rimraf tmp",
     "lint": "prettier --check '**/*.{js,ts,json,yml,md}'",
     "lint:fix": "prettier --write '**/*.{js,ts,json,yml,md}'",
     "dev": "tsc-watch --onFirstSuccess \"npm run dev:make-exec\"",
diff --git a/script/test-template.sh b/script/test-template.sh
index 4204f34..aeb89b4 100755
--- a/script/test-template.sh
+++ b/script/test-template.sh
@@ -5,7 +5,7 @@ set -o pipefail  # Exit when any program execution in a pipeline breaks

 readonly APP="./bin/create-probot-app.js"
 readonly TEMPLATE=$1
-readonly TEST_FOLDER="./test/templates/${TEMPLATE}"
+readonly TEST_FOLDER="./tmp/templates/${TEMPLATE}"
 readonly LOGFILENAME="test.output"
 readonly LOGFILE="./test/templates/${TEMPLATE}/${LOGFILENAME}"
```

Comment (shaftoe) [reply .gitignore:15]: Forgot LOGFILE...

```diff
diff --git a/package.json b/package.json
index 090f7f9..d9911a3 100644
--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
   "scripts": {
     "test": "./script/test-all-templates.sh",
     "test:template": "./script/test-template.sh",
-    "test:clean": "rimraf test",
+    "test:clean": "rimraf tmp",
     "lint": "prettier --check '**/*.{js,ts,json,yml,md}'",
     "lint:fix": "prettier --write '**/*.{js,ts,json,yml,md}'",
     "dev": "tsc-watch --onFirstSuccess \"npm run dev:make-exec\"",
diff --git a/script/test-template.sh b/script/test-template.sh
index 4204f34..2cffa93 100755
--- a/script/test-template.sh
+++ b/script/test-template.sh
@@ -5,9 +5,9 @@ set -o pipefail  # Exit when any program execution in a pipeline breaks

 readonly APP="./bin/create-probot-app.js"
 readonly TEMPLATE=$1
-readonly TEST_FOLDER="./test/templates/${TEMPLATE}"
+readonly TEST_FOLDER="./tmp/templates/${TEMPLATE}"
 readonly LOGFILENAME="test.output"
-readonly LOGFILE="./test/templates/${TEMPLATE}/${LOGFILENAME}"
+readonly LOGFILE="./tmp/templates/${TEMPLATE}/${LOGFILENAME}"

 function create_app() {
   mkdir -p "$TEST_FOLDER"
```

Comment (shaftoe) [reply .gitignore:15]: Or, even better, `readonly LOGFILE="${TEST_FOLDER}/${LOGFILENAME}"`

**Comment by gr2m** [comment]
I prefer `tmp` over `test`, too. Or call it `tmp-test` to be more verbose. `test` is a folder to commonly used in other projects for test code, including other probot repositories, I fear it might be confusing to have that directory being ignored here

File: package.json [hunk 2]
```diff
@@ -12,9 +12,9 @@
   "scripts": {
     "test": "./script/test-all-templates.sh",
     "test:template": "./script/test-template.sh",
-    "test:clean": "rimraf test",
-    "lint": "prettier --check '**/*.{js,ts,json,yml,md}'",
-    "lint:fix": "prettier --write '**/*.{js,ts,json,yml,md}'",
+    "test:clean": "rimraf tmp-test",
+    "lint": "prettier --ignore-path .gitignore --check '**/*.{js,ts,json,yml,md}'",
+    "lint:fix": "prettier --ignore-path .gitignore --write '**/*.{js,ts,json,yml,md}'",
```
Comment (gr2m) [inline package.json:17]: that's a nice hack!

**Review by gr2m — APPROVED** [review]

**Review by shaftoe — APPROVED** [review]

**Pull request merged**